data_handler/handlers/order_books/processing.py

```diff
@@ -0,0 +1,61 @@
+from decimal import Decimal
+
+from db.crud import DBConnector
+
+
+class OrderBookProcessor:
+    def __init__(self, dex, token_a, token_b):
+        """
+        Initialize the order book processor.
+        :param dex: The DEX name to work with.
+        :param token_a: The base token address.
+        :param token_b: The quote token address.
+        """
+        self.dex = dex
+        self.token_a = token_a
+        self.token_b = token_b
+
+    def calculate_price_change(self, price_change_ratio: Decimal) -> Decimal:
+        """
+        Calculate quantity of `token_a` that can be bought to change the price by the given ratio.
+        :param price_change_ratio: Decimal - The price change ratio.
+        :return: Decimal - Quantity that can be traded without moving price outside acceptable bound.
+        """
+        connector = DBConnector()
+        order_book = connector.get_latest_order_book(self.dex, self.token_a, self.token_b)
+        if not order_book:
+            raise ValueError("No order book found for the given DEX and token pair.")
+        current_price = order_book.current_price
+        if price_change_ratio > 1 or price_change_ratio < 0:
+            raise ValueError("Provide valid price change ratio.")
+        if current_price == 0:
+            raise ValueError("Current price of the pair is zero.")
+        min_price = (Decimal("1") - price_change_ratio) * current_price
+        lower_quantity = Decimal("0")
+        for price, quantity in order_book.bids[::-1]:
+            if price >= min_price:
+                lower_quantity += Decimal(quantity)
+            elif price < min_price:
+                break
+        return lower_quantity
```

Comment:
the name of the method is `calculate_price_change`  but it returs `quantity, let's find better naming for it